.github/workflows/build-gradle.yml

```diff
@@ -146,12 +162,17 @@ jobs:
           sed -E -i "s/^\s{4}version = \".+\"/    version = \"${RELEASE_VERSION}\"/g" ${{ inputs.server-path }}/build.gradle.kts
           sed -E -i "s/^version=.+/version=${RELEASE_VERSION}/g" ${{ inputs.server-path }}/gradle.properties
 
-      - name: Server Build
-        uses: gradle/gradle-build-action@v2
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
```

Comment:
https://github.com/gradle/gradle-build-action mentions that `gradle/gradle-build-action` is now deprecated and `gradle/actions/setup-gradle` should be used. When using `gradle/gradle-build-action` caching wasn't working, which I assume is an issue with the action as I was able to use the plan github cache action just fine. After updating to the new action the caching is working.